apps/backend/src/lambdas/helpers.ts

```diff
@@ -0,0 +1,60 @@
+import { parse, subBusinessDays } from 'date-fns';
+import { Ministries } from '../constants';
+
+export const setReconciliationDates = (
+  event: unknown,
+  maxNumDaysToReconcile: number
+) => {
+  const parsedEvent = typeof event === 'string' ? JSON.parse(event) : event;
+
+  if (parsedEvent.reconciliationEventOverride) {
+    return {
+      reconciliationMinDate: parse(
+        parsedEvent.period.from,
+        'yyyy-MM-dd',
+        new Date()
+      ),
+      reconciliationMaxDate: parse(
+        parsedEvent.period.to,
+        'yyyy-MM-dd',
+        new Date()
+      ),
+      currentDate: parse(parsedEvent.period.to, 'yyyy-MM-dd', new Date()),
+      ministry: parsedEvent.program,
+      byPassFileValidity: parsedEvent.byPassFileValidity ?? false,
+    };
+  }
+  const reconciliationMaxDate = subBusinessDays(new Date(), 1);
+  const reconciliationMinDate = subBusinessDays(
+    reconciliationMaxDate,
+    maxNumDaysToReconcile
+  );
+  return {
+    reconciliationMinDate,
+    reconciliationMaxDate: subBusinessDays(new Date(), 1),
+    currentDate: new Date(),
+    ministry: Ministries.SBC,
+    byPassFileValidity: false,
+  };
+};
+
+/**
+ * Extract the file date from the file name
+ */
+export const extractDateFromTXNFileName = (fileName: string): Date => {
```

Comment:
My concern about these functions is that they expect filenames that conform exactly to what we want. The two ways to mitigate that:
- Add an example of the filename in the comment block
- A trycatch / error if it does not conform to what we expect. Same thing if the date is not parseable

Reply:
I've removed the date from filename stuff for bcm. For garms, we don't really have a choice, but I have included a validation to reject the file in the case of an incorrect filename.